frame/vesting/src/lib.rs

```diff
@@ -236,31 +292,67 @@ pub mod module {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// Unlock any vested funds of the origin account.
+		///
+		/// The dispatch origin for this call must be _Signed_ and the sender must have funds still
+		/// locked under this pallet.
+		///
+		/// - `asset`: The asset associated with the vesting schedule
+		/// - `vesting_schedule_ids`: The ids of the vesting schedules to be claimed
+		///
+		/// Emits `Claimed`.
 		#[pallet::weight(<T as Config>::WeightInfo::claim((<T as Config>::MaxVestingSchedules::get() / 2) as u32))]
-		pub fn claim(origin: OriginFor<T>, asset: AssetIdOf<T>) -> DispatchResult {
+		pub fn claim(
+			origin: OriginFor<T>,
+			asset: AssetIdOf<T>,
+			vesting_schedule_ids: VestingScheduleIdSet<
+				T::VestingScheduleId,
+				T::MaxVestingSchedules,
+			>,
+		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			let locked_amount = Self::do_claim(&who, asset)?;
+			Self::do_claim(&who, asset, vesting_schedule_ids)?;
 
-			Self::deposit_event(Event::Claimed { who, asset, locked_amount });
 			Ok(())
 		}
 
+		/// Create a vested transfer.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// - `from`: The account sending the vested funds.
+		/// - `beneficiary`: The account receiving the vested funds.
+		/// - `asset`: The asset associated with this vesting schedule.
+		/// - `schedule_input`: The vesting schedule data attached to the transfer.
+		///
+		/// Emits `VestingScheduleAdded`.
+		///
+		/// NOTE: This will unlock all schedules through the current block.
 		#[pallet::weight(<T as Config>::WeightInfo::vested_transfer())]
 		pub fn vested_transfer(
 			origin: OriginFor<T>,
 			from: <T::Lookup as StaticLookup>::Source,
 			beneficiary: <T::Lookup as StaticLookup>::Source,
 			asset: AssetIdOf<T>,
-			schedule: VestingScheduleOf<T>,
+			schedule_input: VestingScheduleInputOf<T>,
```

Comment:
`schedule_input: VestingScheduleInputOf`so you have var name `shedule_input` which can be pascal cased and prefixed with mod and you get VestingSchdeuleInput. So any naming you pick will bring zero information. Most extenrable naming and neutral is `input`, no prefix.

Reply:
Yeah makes sense. Just renamed it to `vesting_info`, inspired by https://github.com/paritytech/substrate/blob/master/frame/vesting/src/lib.rs